Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -17,15 +17,60 @@ import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
 @MainActor
-class PreplannedMapModel: ObservableObject, Identifiable {
+public class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: any PreplannedMapAreaProtocol
     
+    /// The task to use to take the area offline.
+    private let offlineMapTask: OfflineMapTask?
+    
+    /// The download directory for the preplanned map areas.
+    private let preplannedDirectory: URL?
+    
+    /// The ID of the preplanned map area.
+    private var preplannedMapAreaID: String?
```

Comment:
This property is a variable since the ID may not be set in the init if the `preplannedMapArea.id` is `nil`. Another option is to make PreplannedMapModel initializer failable.